aws_lambda_builders/workflows/java_gradle/DESIGN.md

```diff
@@ -0,0 +1,117 @@
+# Java - Gradle Lambda Builder
+
+## Scope
+
+This package enables the creation of a Lambda deployment package for Java
+projects managed using the Gradle build tool.
+
+For Java projects, the most popular way to create a distribution package for
+Java based Lambdas is to create an "uber" or "fat" JAR. This is a single JAR
+file that contains both the customers' classes and resources, as well as all the
+classes and resources extracted from their dependency JAR's. However, this can
+cause files that have the same path in two different JAR's to collide within the
+uber JAR.
+
+Another solution is to create a distribution ZIP containing the customer's
+classes and resources and include their dependency JARs under a `lib` directory.
+This keeps the customers' classes and resources separate from their
+dependencies' to avoid any file collisions. However, this incurs some overhead
+as the ZIP must be unpacked before the code can run.
+
+To avoid the problem of colliding files, we will choose the second option and
+create distribution ZIP's by default. We can further provide the `--uber-jar`
```

Comment:
Let's try and focus from what the aws-lambda-builders scope is and not options added to the CLI. This seems specific to SAM CLI, and while SAM CLI uses this we need to design for any caller.

Reply:
I'll go ahead and remove this bit for now.